libyul/Utilities.cpp

```diff
@@ -91,9 +91,17 @@ u256 solidity::yul::valueOfNumberLiteral(Literal const& _literal)
 {
 	yulAssert(_literal.kind == LiteralKind::Number, "Expected number literal!");
 
-	std::string const& literalString = _literal.value.str();
-	yulAssert(isValidDecimal(literalString) || isValidHex(literalString), "Invalid number literal!");
-	return u256(literalString);
+	static map<YulString, u256> numberCache;
```

Comment:
How large is this map expected to be? Would making it an `unordered_map` introduce any performance gains, or would it just introduce unnecessary overhead?

Reply:
I'd be curious about this myself - but my guess would be that we don't have *that* many distinct literal values in the yul code for it to help. On the other hand, ``YulString``s being "prehashed" and all, always makes me wonder about it :-).